lib/rules/no-fallthrough.js

```diff
@@ -118,12 +135,17 @@ module.exports = {
                 /*
                  * Checks whether or not there is a fallthrough comment.
                  * And reports the previous fallthrough node if that does not exist.
+                 *
+                 *
                  */
-                if (fallthroughCase && !hasFallthroughComment(fallthroughCase, node, context, fallthroughCommentPattern)) {
-                    context.report({
-                        messageId: node.test ? "case" : "default",
-                        node
-                    });
+
+                if (fallthroughCase && (!hasFallthroughComment(fallthroughCase, node, context, fallthroughCommentPattern))) {
+                    if (!(allowEmptyCase && isEmptyCase(fallthroughCase))) {
+                        context.report({
+                            messageId: node.test ? "case" : "default",
+                            node
+                        });
+                    }
                 }
```

Comment:
```suggestion
                if (fallthroughCase && (!hasFallthroughComment(fallthroughCase, node, context, fallthroughCommentPattern))) {
                    context.report({
                        messageId: node.test ? "case" : "default",
                        node
                    });
                }
```

The added condition seems redundant (logic for the new option is already implemented on line 162).

Reply:
removed redundant logic